### PR TITLE
Fix org dismissal restrictions

### PR DESCRIPTION
**Pull request by AetherDeity**
Addresses https://github.com/PyGithub/PyGithub/issues/1335

### CODE REVIEW

File: github/Repository.py [hunk 1]
```diff
@@ -1518,6 +1518,7 @@ def get_branch(self, branch):
         headers, data = self._requester.requestJsonAndCheck(
             "GET", self.url + "/branches/" + branch
         )
+        data['is_organization'] = self.organization is not None
```
Comment (AetherDeity) [inline github/Repository.py]: This change causes Repository object to be loaded in tests, negating lazy load https://github.com/PyGithub/PyGithub/blob/dd158556a838a180d966925c824100f320a5fcfc/tests/Branch.py#L45

I'm not sure if the test data needs rerecording or if there is a different fix for the resulting failed tests.

Comment (s-t-e-v-e-n-k) [reply github/Repository.py]: Yeah, I think you might want to duck under the covers there and use the underscored attribute.

File: github/Branch.py [hunk 2]
```diff
@@ -196,6 +206,10 @@ def edit_protection(
             or required_approving_review_count is not github.GithubObject.NotSet
         ):
             post_parameters["required_pull_request_reviews"] = {}
+            if self._rawData.get("is_organization", False):
```
Comment (s-t-e-v-e-n-k) [inline github/Branch.py]: Please add a property to Branch that exposes it and check it here.

File: tests/GitRelease.py [hunk 3]
```diff
@@ -51,9 +51,8 @@ def setUp(self):
         with open(self.content_path, "w") as zip_content:
             zip_content.write("Pedro for president.")
 
-        artifact = zipfile.ZipFile(self.artifact_path, "w")
-        artifact.write(self.content_path)
-        artifact.close()
+        with zipfile.ZipFile(self.artifact_path, "w") as artifact:
+            artifact.write(self.content_path)
```
Comment (s-t-e-v-e-n-k) [inline tests/GitRelease.py]: This is a nice change, but can you split it out to a separate PR?

File: github/Branch.py [hunk 4]
```diff
@@ -71,6 +71,13 @@ def protected(self):
         """
         return self._protected.value
 
+    @property
```
Comment (s-t-e-v-e-n-k) [inline github/Branch.py]: I mean, I get why you're adding it, but we also don't check it anywhere at all ... Not even tests.

File: github/Repository.py (hunk 1 above)
Comment (AetherDeity) [reply github/Repository.py]: I did look at doing that, but as I understand the lazy loading, the organization key would not exist before completing the object. The result would be a lazy loaded Repository would always look like a user repo.

File: github/Branch.py (hunk 2 above)
Comment (AetherDeity) [reply github/Branch.py]: I wasnt sure if it ought to be exposed as it isn't a native GitHub attribute, but I can certainly do that.

File: github/Branch.py (hunk 4 above)
Comment (AetherDeity) [reply github/Branch.py]: As an attribute returned by GitHub, I thought it made sense to expose. I'm not sure if you are asking for this to be removed or for tests to check it.

File: tests/GitRelease.py (hunk 3 above)
Comment (AetherDeity) [reply tests/GitRelease.py]: Can do
